service/controller/resource/ipam/azure_machinepool_network_range_getter.go

```diff
@@ -38,6 +46,11 @@ func NewAzureMachinePoolNetworkRangeGetter(config AzureMachinePoolNetworkRangeGe
 // node pool subnet is getting its IP address range from all available address
 // ranges in the tenant cluster virtual network.
 func (g *AzureMachinePoolNetworkRangeGetter) GetNetworkRange(ctx context.Context, obj interface{}) (net.IPNet, error) {
+	g.logger.LogCtx(
+		ctx,
+		"level", "debug",
+		"message", "getting tenant cluster's VNet range from which the node pool subnet will be allocated")
+
```

Comment:
If we are adding a new error to gracefully handle not having a `CidrBlock` just yet, I think we should do the same for the case when the cluster label is missing and `GetAzureClusterFromMetadata()` returns an error.
We can do it in another PR if you want.

Reply:
:thinking: not sure if those two cases are the same
- `CidrBlock` is missing because it's still not allocated, when a new cluster is being created that happens by design
- cluster label, when it's missing, means that it is not set, while it has to be set according to the spec - IMHO this should be solved by proper defaulting, so we don't have to check everywhere in code if something that should be set is set or not, since a missing label is an error in the CR

I think should have a chat about validation and defaulting :)

WDYT?

Reply:
We have a controller that adds the label as well, so eventually it will be there. If we want to make it mandatory to have the label, then errorring out is fine I guess.

Reply:
I believe we decided that it's mandatory to have the label per spec.